index.js

```diff
@@ -33,11 +34,62 @@ Toolkit.run(async tools => {
     body: env.renderString(body, templateVariables),
     title: env.renderString(attributes.title, templateVariables)
   }
-
   tools.log.debug('Templates compiled', templated)
-  tools.log.info(`Creating new issue ${templated.title}`)
+
+  if (updateExisting) {
+    let existingIssue
+    tools.log.info(`Fetching issues issues with title "${templated.title}"`)
+    try {
+      existingIssue = await tools.github.issues.listForRepo({
+        ...tools.context.repo,
+        state: 'open'
+      }).find(existingIssue => {
+        return existingIssue.title === attributes.title && existingIssue.state === 'open'
+      })
+    } catch (err) {
+      const errorMessage = 'An error occurred while retrieving existing issues.'
+      tools.log.error(errorMessage)
+      tools.log.error(err)
+
+      // The error might have more details
+      if (err.errors) tools.log.error(err.errors)
+
+      // Exit with a failing status
+      core.setFailed(errorMessage + '\n\n' + err.message)
+      tools.exit.failure()
+    }
```

Comment:
I appreciate that you added some error handling here, but I think its more robust than necessary - how about just:

```suggestion
    } catch (err) {
      tools.exit.failure(err)
    }
```